### ISSUE TITLE
Multiple tables issue

### ISSUE THREAD

**Issue by apperside** [AllenFang/react-bootstrap-table#1428]
Hi,
I am playing with this amazing library and I have an issue m which happens only when I use 2 (or more) tables in a page:

An you can see, on both tables I use a custom editor, and when I click on a cell on the first table to edit it, and then on the custom editor I click cancel, my chrome debugger stops in the point shown [here](https://docs.google.com/drawings/d/1zlYARF9UuDMyL9XRjwYSs9PeHY2z50kescVN5oUH4jw/edit?usp=sharing) (hope the developer understands which point it is from the screenshot, otherwise I can give all the needed help to get it out).

Let's say I have 10 records on first table, and 5 on the second one and I click on the 7th row of the first table: it' s quite funny , but inpecting the variables on the breakpoint, it seem that the library's  function getCurrentDisplayData() returns the data from the the second table (which has less rows than the first one), and it breaks on line 62154 because currentDisplayData[rowIndex] is undefined, since rowIndex is 7, but currentDisplayData array has only 5 elements (the ones from the second table)
```

const cellEditProp: CellEdit = {
    mode: 'click'
};
const categorySelectionOptions = {
    mode: 'radio'
}
const subCategorySelectionOptions = {
    mode: 'checkbox'
}
 createDefaultTextEditor(onUpdate: any, props: any) {
        return <DefaultTextEditor onUpdate={onUpdate} {...props} />;
    }

getDefaultTextEditor(fieldName: string) {
        return {
            getElement: this.createDefaultTextEditor,
            customEditorParameters: { fieldName: fieldName }
        }
    }

type props = {
    load?: (load: boolean) => void,
    row: MyBaseModel,
    fieldName: string,
    onUpdate(newValue: string),
    showError(title: string, message: string),
    editorClass: string
}

type state = {
    object: MyBaseModel,
    value: string,
}
export default class DefaultTextEditor extends React.Component<props, state> {

    inputRef: HTMLInputElement
    constructor(props: props) {
        super(props);
        console.log("DefaultTextEditor props", props);
        this.updateData = this.updateData.bind(this);
        this.cancel = this.cancel.bind(this);
        this.state = {
            object: props.row,
            value: props.row[props.fieldName] || ""
        };
    }

    focus() {
        // this.refs.inputRef.focus();
    }

    updateData() {
                this.props.onUpdate(this.state.value);
    }

    cancel() {
        this.props.onUpdate(this.props.row[this.props.fieldName]);
    }

    render() {
        return (
            <div>
                <input
                    className={(this.props.editorClass || '') + ' form-control editor edit-text'}
                    style={{ display: 'inline', width: '50%' }}
                    type='text'
                    onKeyUp={(e) => {
                        if (e.keyCode == 13) {
                            this.setState({
                                value: e.currentTarget.value
                            }, () => {
                                this.updateData();
                            });
                        }
                    }}
                    value={this.state.value}
                    onChange={(ev) => {
                        this.setState({ value: ev.currentTarget.value });
                    }} />
                <div style={{ marginTop: 10 }}>
                    <button
                        style={{ position: 'relative', float: 'left' }}
                        className='btn btn-info btn-xs textarea-save-btn'
                        onClick={this.updateData}>
                        SALVA
                    </button>
                    <button
                        style={{ marginLeft: 10, position: 'relative', float: 'left' }}
                        className='btn btn-info btn-xs textarea-save-btn'
                        onClick={this.cancel}>
                        ANNULLA
                    </button>
                </div>
            </div>
        );
    }
}

<BootstrapTable data={this.state.categories || []}
                        deleteRow
                        cellEdit={cellEditProp}
                        selectRow={this.state.categorySelectionOptions}
                        insertRow>
                        <TableHeaderColumn dataField='objectId'
                            isKey={true}
                            hiddenOnInsert
                            hidden>Product ID</TableHeaderColumn>
                        <TableHeaderColumn dataField='name'
                            filter={{ type: 'TextFilter', delay: 250, placeholder: 'enter a name' }}
                            customEditor={getDefaultTextEditor("name")}>
                            Name</TableHeaderColumn>
                        <TableHeaderColumn dataField='code'
                            customEditor={this.getDefaultTextEditor("code")}>
                            Category code
                        </TableHeaderColumn>
                    </BootstrapTable>


<BootstrapTable data={this.state.subCategories || []}
                        deleteRow
                        cellEdit={cellEditProp}
                        selectRow={this.state.subCategorySelectionOptions}
                        insertRow>
                        <TableHeaderColumn dataField='objectId'
                            isKey={true}
                            hiddenOnInsert
                            hidden>Product ID</TableHeaderColumn>
                        <TableHeaderColumn dataField='name'
                            filter={{ type: 'TextFilter', delay: 250, placeholder: 'enter a name' }}
                            customEditor={getDefaultTextEditor("name")}>
                            Name</TableHeaderColumn>
                        <TableHeaderColumn dataField='code'
                            customEditor={this.getDefaultTextEditor("code")}>
                            Sub Category Code
                        </TableHeaderColumn>
                    </BootstrapTable>
```

**Comment by AllenFang** [comment]
@apperside, I'll check this, but need some time, I'm busy in recently.

**Comment by pengi** [comment]
This affects me too. I have a list of objects, each including an editable table of properties. The workaround right now would be to only show the properties table for one of the objects.

Otherwise, it works great!

**Comment by pengi** [comment]
Oh, while trying to fix it with having a couple of tables non-editable and one editable, I've ended up getting it to fully work half way.

The thing is that I've changed my cellEditProp to a function, which means I don't reuse the object between tables. It seems that BootstrapTable modifies the object passed as cellEdit and not a clone of the object. The problem is probably here:

https://github.com/AllenFang/react-bootstrap-table/blob/822b9bc9104551a6ecfdb7149487f1e4efb7a51d/src/BootstrapTable.js#L333
